internal/cmd/destinations.go

```diff
@@ -56,6 +57,12 @@ func newDestinationsListCmd(cli *CLI) *cobra.Command {
 					return err
 				}
 				cli.Output(string(jsonOutput))
+			case "yaml":
```

Comment:
Any way we could have a re-usable function for the output formatting to reduce the repetition in the cli output code here?

Reply:
Mentioned in the previous change. There should be a `Formatter` interface which `JSON`/`YAML`/`TableFormat` implements. The result is passed to `cli.Output` for printing. This will allow more formats to be added with minimal impact on the existing code base

Reply:
For the actual function, not sure how this can be achieved, could I have an example? 

The couple of commands here are all `list`, so it expects similar outputs, but different functions will not use the same `something.Items` but rather be completely different.

Reply:
resolved & done

Reply:
I put together an example of one option here: https://github.com/infrahq/infra/compare/cli-9...dnephin/formatter-example

But let's look into that in a follow up PR.